source/solvers/heat_transfer_assemblers.cc

```diff
@@ -551,7 +549,115 @@ HeatTransferAssemblerViscousDissipation<dim>::assemble_rhs(
+HeatTransferAssemblerViscousDissipationVOF<dim>::assemble_rhs(
+  HeatTransferScratchData<dim> &scratch_data,
+  StabilizedMethodsCopyData &   copy_data)
+{
+  const unsigned int n_q_points = scratch_data.n_q_points;
+  const unsigned int n_dofs     = scratch_data.n_dofs;
+
+  const std::vector<double> &density   = scratch_data.density;
+  const std::vector<double> &viscosity = scratch_data.viscosity;
+
+  double viscous_dissipation_coefficient(0.);
+  //  const std::vector<double> &viscous_dissipation_coefficient =
+  //    scratch_data.viscous_dissipation_coefficient;
+
+  // Time steps and inverse time steps which is used for stabilization
+  // constant
+  std::vector<double> time_steps_vector =
+    this->simulation_control->get_time_steps_vector();
+
+  // Copy data elements
+  auto &local_rhs = copy_data.local_rhs;
+
+
+  //  // Coefficient used to consider viscous dissipation in
+  //  // a given fluid only
+  //  if (this->viscous_dissipation_indicator ==
+  //      Parameters::DissipationIndicator::fluid1)
+  //    {
+  //      // if phase = 0, no viscous dissipation
+  //      // if phase = 1, maximum viscous dissipation
+  //      this->viscous_dissipation_coefficient[q] =
+  //        this->phase_values[q];
+  //    }
+  //  else if (this->viscous_dissipation_indicator ==
+  //           Parameters::DissipationIndicator::fluid0)
+  //    {
+  //      // if phase = 1, no viscous dissipation
+  //      // if phase = 0, maximum viscous dissipation
+  //      this->viscous_dissipation_coefficient[q] =
+  //        1. - this->phase_values[q];
+  //    }
+  //  else
+  //    {
+  //      // maximum viscous dissipation everywhere
+  //      this->viscous_dissipation_coefficient[q] = 1.;
+  //    }
+
+  // assembling right hand side
+  for (unsigned int q = 0; q < n_q_points; ++q)
+    {
+      // Gather physical properties in case of mono fluids simulations (to be
+      // modified by cell in case of multiple fluids simulations)
+      const double dynamic_viscosity = viscosity[q] * density[q];
+
+      // Store JxW in local variable for faster access
+      const double JxW = scratch_data.fe_values_T.JxW(q);
+
+      const auto velocity_gradient_fd =
+        scratch_data.velocity_gradient_values[q];
+
+      // Manage viscous dissipation application on specified fluid
+      const double phase_value_q = scratch_data.phase_values[q];
+      if (this->viscous_dissipation_indicator ==
+          Parameters::DissipationIndicator::fluid1)
+        {
+          // if phase = 0, no viscous dissipation
+          // if phase = 1, maximum viscous dissipation
+          viscous_dissipation_coefficient = phase_value_q;
+        }
+      else if (this->viscous_dissipation_indicator ==
+               Parameters::DissipationIndicator::fluid0)
+        {
+          // if phase = 1, no viscous dissipation
+          // if phase = 0, maximum viscous dissipation
+          viscous_dissipation_coefficient = 1. - phase_value_q;
+        }
+      else
```

Comment:
I know you have the throw on the parameter, but I would use "else if" here just in case.